src/gpu/align_z_to_other.cu

```diff
@@ -0,0 +1,208 @@
+            const int other_x0 = static_cast<int>(other_pixel[0] + 0.5f);
+            const int other_y0 = static_cast<int>(other_pixel[1] + 0.5f);
+
+            // Map the bottom-right corner of the depth pixel onto the other image
+            depth_pixel[0] = depth_x + 0.5f;
+            depth_pixel[1] = depth_y + 0.5f;
+            rs_deproject_pixel_to_point(depth_point, &depth_intrin, depth_pixel, depth, inv_fx, inv_fy);
+            rs_transform_point_to_point(other_point, &depth_to_other, depth_point);
+            rs_project_point_to_pixel(other_pixel, &other_intrin, other_point);
+            const int other_x1 = static_cast<int>(other_pixel[0] + 0.5f);
+            const int other_y1 = static_cast<int>(other_pixel[1] + 0.5f);
+
+            if (other_x0 < 0 || other_y0 < 0 || other_x1 >= other_intrin.width ||
+                other_y1 >= other_intrin.height)
+                return;
+            // Transfer between the depth pixels and the pixels inside the rectangle on the other image
+            for (int y = other_y0; y <= other_y1; ++y)
+                for (int x = other_x0; x <= other_x1; ++x)
+                    transfer_pixel(depth_pixel_index, y * other_intrin.width + x, z_pixels_gpu, out_pixels_gpu);
+        }
+    }
+
+
+    class CudaStreamWrapper {
+    private:
+        cudaStream_t stream;
+    public:
+        CudaStreamWrapper() {
+            cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking);
+            ROS_INFO_THROTTLE(60, "GPU Stream Created");
+
+        }
+
+        ~CudaStreamWrapper() {
+            cudaStreamDestroy(stream);
+            ROS_INFO_THROTTLE(60, "GPU Stream Destroyed");
+
+        }
+
+        operator cudaStream_t&(){
+            return stream;
+        }
+
+
+    };
+
+    __host__ bool align_z_to_other(rsimpl::byte * z_aligned_to_other, const uint16_t * z_pixels, float z_scale, const rs_intrinsics & z_intrin, const rs_extrinsics & z_to_other, const rs_intrinsics & other_intrin)
+    {
+        ROS_INFO_THROTTLE(60, "GPU NOT DISABLED");
+
+        int deviceCount = 0;
+        auto cudaCallErrorStatus = cudaGetDeviceCount(&deviceCount);
+        if (cudaSuccess != cudaCallErrorStatus) {
+            ROS_ERROR("Failed to obtain number of GPUs. %s %s", cudaGetErrorName(cudaCallErrorStatus), cudaGetErrorString(cudaCallErrorStatus));
+            return false;
+        }
+        if (deviceCount == 0) {
+            ROS_WARN("NO ENABLED GPUs PRESENT");
+            return false;
+        }
+
+
+        // ignore this error. technically the blocking sync can only be set once prior to device initialization. once
+        // that initial setting is done, subsequent settings result in an error until device is restarted
+        cudaCallErrorStatus = cudaSetDeviceFlags(cudaDeviceScheduleBlockingSync);
+
+        auto out_z = (uint16_t *)(z_aligned_to_other);
+
+        auto cudaDeleterUnsignedInt = [](unsigned int *ptr) {ROS_INFO_THROTTLE(60, "CUDA FREE INVOKED"); cudaFree(ptr);};
```

Comment:
Maybe wrap cudaMalloc/cudaFree into objects?  Definitely should be debug level log messages OR remove before we merge.